pumpkin-protocol/src/uuid.rs

```diff
@@ -11,3 +11,9 @@ impl Serialize for UUID {
         serializer.serialize_bytes(self.0.as_bytes())
     }
 }
+
+impl UUID {
+    pub fn new(data: &[u8]) -> Self {
+        Self(uuid::Uuid::new_v3(&uuid::Uuid::NAMESPACE_DNS, data))
+    }
```

Comment:
This code should be the equivalent to Java's `UUID::nameUUIDFromBytes`. I think its wrong to put this as an new

Reply:
Hmm. I agree somewhat. This wrapper is overly generic for a "UUID", but this is it's only usage right now. Do you believe there's a better suited name for it?

Reply:
Imo using `new` is not bad in this situation, but you should at least add a doc comment explicitly saying that it is using `uuid::Uuid::NAMESPACE_DNS`, since that might be misleading otherwise.

Reply:
I don't believe a Namespaced uuidv3 UUID is the sole UUID type in pumpkin. It should change regardless of the mechanism. Considering that this is only a newtype wrapper around an existing UUID type you should either completely encapsulate the underlying UUID type or not encapsulate it at all. Otherwise there's not much gained by a newtype wrapper here.

Reply:
Yeah alr, but just use doc comment if this stays as it is